client/verta/verta/_internal_utils/_utils.py

```diff
@@ -528,7 +528,15 @@ def raise_for_http_error(response):
         curr_time = timestamp_to_str(now(), utc=True)
         time_str = " at {} UTC".format(curr_time)
 
-        reason = six.ensure_str(response.text.strip())
+        try:
+            reason = body_to_json(response)['message']
+        except (ValueError,  # not JSON response
+                TypeError,  # body_to_json return a single string
```

Comment:
@convoliution apparently body_to_json could return a string 🤷Specifically, response.json() returns `LIMIT_ENDPOINT_NUMBER: currently at maximum number of endpoints` when we try to create a second endpoint 🤔

Reply:
Oh, that complicates things.
In that case, could you do me a favor and split this `try-except` into multiple checks?
It was already a shaky idea when I tried catching two error types simultaneously—trying to catch 3 is really prone to issues down the line

Something like (pseudocode)
```
try:
    reason = body_to_json()
except ValueError:
    raise

if reason is dict and 'message' in reason:
    reason['message']

# etc...

Reply:
Fixed. See [latest commit](https://github.com/VertaAI/modeldb/pull/1650/commits/5ad1c0319dba02d7a492e8cfffe50e0babfbdf14).